File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int left = 0, right = 0;
        while(right < nums.length){
            if(nums[left] != 0){
                left++;
            } else if(nums[right] != 0){
                nums[left] = nums[right];
                nums[right] = 0;
                left++;
            }
            right++;
        }
    }
}

File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = None
        for j in range(len(nums)):
            if i is None:
                if nums[j] == 0:
                    i = j
            elif nums[j] != 0:
                nums[i], nums[j] = nums[j], 0
                i += 1